src/components/FormStep.js

```diff
@@ -399,12 +515,25 @@ const FormStep = ({form, submission, onLogicChecked, onStepSubmitted, onLogout})
       });
     } catch (e) {
       dispatch({type: 'LOGIC_CHECK_INTERRUPTED', payload: {canSubmit}});
+
       if (e.name !== 'AbortError') {
         throw e; // re-throw on unexpected errors
       }
     }
   };
 
+  /**
+   * A form has been submitted.
+   * @see {onReactSubmit} We wrap the submit so that we control our own submit buttonLOGIC_CHECK_DONE
+   *
+   * During evaluation, the following actions may be dispatched:
+   *
+   *   - `NAVIGATE` Immediately (if submission is available).
+   *   - `ERROR` When an error occurred while loading the form step configuration.
```

Comment:
this really doesn't make things clearer for me looking just at the docstring - I actually have no idea at the moment what it means unless I go looking at the code anyway. Should probably be inline comments.

Reply:
I'd much rather leave them in as I'm going to need them in the near future and I expect code to be largely rewritten.

Reply:
Discussed: keep for now, may be removed after refactor.